File: app/src/main/java/net/mmho/photomap2/PhotoListActivity.java
package net.mmho.photomap2;

import android.content.pm.PackageManager;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v4.app.Fragment;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.View;
import android.view.animation.AlphaAnimation;
import android.view.animation.Animation;
import android.widget.ProgressBar;


public class PhotoListActivity extends AppCompatActivity implements ProgressChangeListener{

    public static final int PERMISSIONS_REQUEST = 1;
    private ProgressBar progressBar;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_photo_list);
        setSupportActionBar((Toolbar) findViewById(R.id.toolbar));
        progressBar = (ProgressBar) findViewById(R.id.progress);
    }

    @Override
    public void onRequestPermissionsResult(int requestCode,@NonNull String[] permissions,
                                           @NonNull int[] grantResults) {
        Fragment fragment = getSupportFragmentManager().findFragmentById(R.id.fragment);
        switch (requestCode){
        case PERMISSIONS_REQUEST:
            boolean granted = grantResults.length > 0
                && grantResults[0] == PackageManager.PERMISSION_GRANTED;
            if(fragment!=null){
                ((PhotoListFragment)fragment).grantedPermission(granted);
            }
        }
    }

    @Override
    public void showProgress(int progress){
        progressBar.setProgress(progress);
        progressBar.setVisibility(View.VISIBLE);
    }

    @Override
    public void endProgress(){
        progressBar.setProgress(progressBar.getMax());
        AlphaAnimation fadeout;
        fadeout = new AlphaAnimation(1,0);
        fadeout.setDuration(300);
        fadeout.setFillAfter(true);
        fadeout.setAnimationListener(new Animation.AnimationListener() {
            @Override
            public void onAnimationStart(Animation animation) {

            }

            @Override
            public void onAnimationEnd(Animation animation) {
                progressBar.setVisibility(View.GONE);
                progressBar.setAnimation(null);
            }

            @Override
            public void onAnimationRepeat(Animation animation) {

            }
        });
        progressBar.startAnimation(fadeout);

    }


}

File: app/src/main/kotlin/net/mmho/photomap2/PhotoListActivity.kt
package net.mmho.photomap2

import android.content.pm.PackageManager
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.View
import android.view.animation.AlphaAnimation
import android.view.animation.Animation
import kotlinx.android.synthetic.main.activity_photo_list.*


class PhotoListActivity : AppCompatActivity(), ProgressChangeListener {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_photo_list)
        setSupportActionBar(toolbar)
    }

    override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<String>,
                                            grantResults: IntArray) {
        val fragment = supportFragmentManager.findFragmentById(R.id.fragment)
        when (requestCode) {
            PERMISSIONS_REQUEST -> {
                val granted = grantResults.size > 0 && grantResults[0] == PackageManager.PERMISSION_GRANTED
                (fragment as PhotoListFragment)?.grantedPermission(granted)
            }
        }
    }

    override fun showProgress(progress: Int) {
        this.progress.progress = progress
        this.progress.visibility = View.VISIBLE
    }

    override fun endProgress() {
        progress.progress = progress.max
        val fadeout = AlphaAnimation(1f, 0f)
        fadeout.duration = 300
        fadeout.fillAfter = true
        fadeout.setAnimationListener(object : Animation.AnimationListener {
            override fun onAnimationStart(animation: Animation) {

            }

            override fun onAnimationEnd(animation: Animation) {
                progress.visibility = View.GONE
                progress.animation = null
            }

            override fun onAnimationRepeat(animation: Animation) {
            }
        })
        progress.startAnimation(fadeout)

    }

    companion object {
        val PERMISSIONS_REQUEST = 1
    }
}
